ludwig/data/preprocessing.py

```diff
@@ -1205,6 +1208,43 @@ def build_data(
     return proc_cols
 
 
+def balance_data(dataset_df, features, preprocessing_parameters, backend):
+    target = None
+    target_count = 0
+    for feature in features:
+        if feature[OUTPUT_FLAG]:
+            target = feature
+            target_count += 1
+
+    if target_count > 1:
+        raise ValueError("Class balancing is only available for single target datasets")
+    if target[TYPE] != BINARY:
+        raise ValueError("Class balancing is only supported for binary output types")
+
+    if backend.df_engine.partitioned:
+        majority_class = backend.df_engine.compute(dataset_df[target[PROC_COLUMN]].value_counts()).idxmax()
+        minority_class = backend.df_engine.compute(dataset_df[target[PROC_COLUMN]].value_counts()).idxmin()
+    else:
+        majority_class = dataset_df[target[PROC_COLUMN]].value_counts().idxmax()
+        minority_class = dataset_df[target[PROC_COLUMN]].value_counts().idxmin()
+    majority_df = dataset_df[dataset_df[target[PROC_COLUMN]] == majority_class]
+    minority_df = dataset_df[dataset_df[target[PROC_COLUMN]] == minority_class]
+
+    if preprocessing_parameters["oversample_minority"]:
```

Comment:
I felt the same way, but interestingly enough, I read through the sampling post travis put in the ml channel last night and someone mentioned a point about using undersampling for both cost reduction purposes and also to combat overfitting since oversampling too much could cause some overfitting. I agree that 95-99% of the time the users will want oversampling, but I think it's nice to have in there in the off chance someone wants to use it for those reasons. WDYT?

Reply:
Great idea, I've added in that check!